pkg/interactive-installer/Dockerfile

```diff
@@ -0,0 +1,25 @@
+# syntax=docker/dockerfile-upstream:1.5.0-rc2-labs
+FROM lfedge/eve-alpine:9fb9b9cbf7d90066a70e4704d04a6fe248ff52bb AS build
+SHELL ["/bin/ash", "-eo", "pipefail", "-c"]
+
+ENV BUILD_PKGS curl build-base git ncurses-dev musl-dev python3
+ENV PKGS alpine-baselayout ncurses-dev smartmontools bash mtools strace util-linux e2fsprogs e2fsprogs-extra jq
+
+RUN eve-alpine-deploy.sh
+ENV RUSTUP_HOME=/usr/local
+ENV CARGO_HOME=/usr/local/cargo
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain none
```

Comment:
There is really no way to fetch the binary packages with ADD?

We try to not allow network access for containers during building as much as possible. Quoting explanation from @eriknordmark:
> We build as much as possible that way to ensure that any source is done using ADD commands so that they make it into the SPDX SBoM file.
> If networking is enabled then someone might later add some curl/wget/git pull way to add source which would not be included in the SBoM.

Original thread: https://github.com/lf-edge/eve/pull/3916#discussion_r1652951132

Reply:
Oh yeah, I missed this one. Nice catch @rene and thank you.

I will add to it in the build.yml

Reply:
Added.

Reply:
If possible, we should just have the rust installed as part of `eve/alpine`.

Reply:
If we cannot do it as part of `eve/alpine`, then we should do something like:

```dockerfile
ADD https://static.rust-lang.org/dist/rust-1.79.0-x86_64-unknown-linux-musl.tar.xz
RUN    # do the install command here based on https://forge.rust-lang.org/infra/other-installation-methods.html#standalone-installers
```

That both uses `ADD` and points to a very specific release; we cannot have the general `curl | sh` which is non-reproducible.

Reply:
I think this is partially addressed. The `network: yes` has been addressed, it looks like it all works now with `network: no`, so thank you for that @jsfakian .

For the other part, can we avoid having to install rust here explicitly? This may be the first package built on rust, but it will not be the last. We should not have to start tracking down each place we add it.

Can we have a separate PR adding rust to eve-alpine, and then just add it as part of `BUILD_PKGS`? eve-alpine is currently mostly based on alpine:3.16 (I think someone is looking at updating it), but we have the mechanism to pull in from other repositories, e.g. edge, [here](https://github.com/lf-edge/eve/tree/master/pkg/alpine/mirrors/edge). 

* Rust in alpine:3.16 is [1.60.0](https://pkgs.alpinelinux.org/packages?name=rust&branch=v3.16&repo=&arch=x86_64&maintainer=)
* Rust in alpine:edge is [1.80.1](https://pkgs.alpinelinux.org/packages?name=rust&branch=edge&repo=&arch=x86_64&maintainer=)

Reply:
Let's do this if we can, and not add more technical debt by having each package that needs rust install it.

Reply:
I opened #4145 to add rust to our base eve-alpine. If it works and goes in quickly, you will be able to just use a more recent version of `FROM lfedge/eve-alpine` and then add `rust` to the `BUILD_PKGS` and have it.

Reply:
I have already tried that, but the code does not compile. It needs a newer version of Rust. If we upgrade to Alpine 3.20 it will be fine. I can try to change the code to make it work though.